projects/schema-form/src/lib/model/formproperty.ts

```diff
@@ -174,6 +178,38 @@ export abstract class FormProperty {
     }
   }
 
+  /**
+   * Making use of the expression compiler for the <code>visibleIf</code> condition
+   */
+  private __evaluateVisibilityIf(
+    sourceProperty: FormProperty,
+    targetProperty: FormProperty,
+    dependencyPath: string,
+    value: any = '',
+    expression: string = ''): boolean {
+    try {
+      let valid = false
+      if (expression.indexOf('$ANY$') !== -1) {
+        valid = value.length > 0;
+      } else if ((expression||[]).toString().indexOf('$EXP$') === 0) {
+        // since visibleIf condition values are an array... we must do this
+        const expArray = Array.isArray(expression) ? expression : (expression ? [expression] : [])
+        for (const expString of expArray) {
```

Comment:
I am not sure I get it.
If it is always an array with only one item, then we should only test `expArray[0]`.
And if it might contain more than one item, then we should do:
```
expArray.some(expString => {
...
})
```
Because with this loop here, `valid` will just have the value for the last item, so it seems wrong, no?

Reply:
You'r right, it should return the first valid occurrence.
Thx